client/src/components/Tickets/TicketModal.js

```diff
@@ -34,22 +36,57 @@ const formSchema = Yup.object().shape({
   reply: Yup.string().required('Required')
 });
 const Attachments = ({ attachments, ticketId }) => {
+  const infiniteScrollCallback = useCallback(() => {});
+  const noDataText = 'No attachments to display.';
+  const json = attachments.map(function attachmentAcessor(x) {
+    return {
+      attachment_id: x[0],
+      attachment_name: x[1]
+    };
+  });
+
+  const columns = [
+    {
+      Header: 'Attached Files',
+      accessor: 'attachment_name',
+      className: 'attachment-title',
+      Cell: el => (
+        <span
+          title={el.value}
+          id={`attachment${el.row.index}`}
+          className="attachment__name"
+        >
+          {el.value}
+        </span>
+      )
+    },
+    {
+      Header: '',
+      className: 'link',
+      accessor: 'attachment_id',
+      Cell: el => (
+        <a
+          href={`/api/tickets/${ticketId}/attachment/${el.value}`}
+          className="link"
+          target="_blank"
+          rel="noreferrer noopener"
+          key={el.value}
+        >
+          Download
```

Comment:
This looks great! My only comment is perhaps we should shrink the size of the `Download` element to be proportional to its contents, that way we can see more of the attachment name:
<img width="446" alt="Screen Shot 2021-11-12 at 9 59 22 AM" src="https://user-images.githubusercontent.com/20326896/141487654-91dfbf54-4795-4733-b97d-c163f1c1cdca.png">

Reply:
@rstijerina 
![Screen Shot 2021-11-12 at 11 04 37 AM](https://user-images.githubusercontent.com/34716439/141506108-99c52d47-b1d8-4933-91f4-5573f2cd75f4.png)
![Screen Shot 2021-11-12 at 11 43 19 AM](https://user-images.githubusercontent.com/34716439/141511251-2c327b93-73e5-4546-9707-32310b7507da.png)
so if I use either `inline-block or inline-flex or inline` it will reduce the size but the attachment title cell is still fixed bc the cell size is seem fixed from InfiniteScrollTable.js (also any inline method causing the Border Discrepancy). If I try to change  InfiniteScrollTable.js it will affect all other tables that use it. Maybe this could be a separate task or a follow-on task?

Reply:
Portal tables are not well-tamed yet. I like the careful thought process. I also suggest a new ticket. Here's some ideas for solutions.

<details>
<summary>Ideas</summary>

**A.** You can set percentage widths for columns via table cells (and change as necessary for window width). This takes a lot of code and some time, but is the current solution for this problem on other tables.

- [example code](https://github.com/TACC/Core-Portal/blob/6ff1e57/client/src/components/Jobs/Jobs.scss)

**B.** You can override the `table-layout: fixed;` for just the ticket modal's table, but you'll need to find a way around the [the consequences](https://github.com/TACC/Core-Portal/blob/a1462f1/client/src/components/_common/InfiniteScrollTable/InfiniteScrollTable.scss#L4-L8).

- <details>
    <summary>Details</summary>

    You can see/test the consequences by putting a very long filename in the filename cell. ![Table-Width Auto Long Cell](https://user-images.githubusercontent.com/62723358/141818917-3c5ae72e-6703-4251-ab49-b69591d5c127.png)

    </details>

**C.** Change the display property of table rows (and maybe table body) to overwrite the native layout of a table.

</details>